### PR TITLE
Cleanup WC_ constants

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup WC_ constants

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2257)

### CODE REVIEW

File: src/Common/src/NativeMethods.cs [hunk 1]
```diff
@@ -987,15 +987,7 @@ public static User32.WindowMessage WM_UIUNSUBCLASS
 
         public const string TOOLTIPS_CLASS = "tooltips_class32";
```
Comment (gpetrou) [inline src/Common/src/NativeMethods.cs]: Doesn't this belong to WC_ as well?

Comment (hughbe) [reply src/Common/src/NativeMethods.cs]: Nope, https://github.com/tpn/winsdk-10/blob/9b69fd26ac0c7d0b83d378dba01080e93349c2ed/Include/10.0.16299.0/um/CommCtrl.h#L2209-L2210

Not sure what do with this constant?

File: src/Common/src/Interop/ComCtl32/Interop.WC.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+internal static partial class Interop
+{
+    internal static partial class ComCtl32
+    {
+        public static class WC
```
Comment (RussKie) [inline src/Common/src/Interop/ComCtl32/Interop.WC.cs]: These are windows classes, but not all of windows classes start with "WC_": https://docs.microsoft.com/en-us/windows/win32/controls/common-control-window-classes
Perhaps have it like this?
```cs
        public static class WindowClasses
        {
			public const string TOOLTIPS_CLASS = "tooltips_class32";

            public const string WC_DATETIMEPICK = "SysDateTimePick32";
            public const string WC_LISTVIEW = "SysListView32";
            public const string WC_MONTHCAL = "SysMonthCal32";
            public const string WC_PROGRESS = "msctls_progress32";
            public const string WC_STATUSBAR = "msctls_statusbar32";
            public const string WC_TOOLBAR = "ToolbarWindow32";
            public const string WC_TRACKBAR = "msctls_trackbar32";
            public const string WC_TREEVIEW = "SysTreeView32";
            public const string WC_TABCONTROL = "SysTabControl32";
        }
```

This also will address @gpetrou question https://github.com/dotnet/winforms/pull/2257/files#r341194690

File: src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs [hunk 3]
```diff
@@ -2,22 +2,160 @@
...
+            Assert.Null(createParams.Param);
+            Assert.Equal(0x5601000B, createParams.Style);
+            Assert.Equal(75, createParams.Width);
+            Assert.Equal(0, createParams.X);
+            Assert.Equal(0, createParams.Y);
+            Assert.Same(createParams, control.CreateParams);
+            Assert.False(control.IsHandleCreated);
+        }
+
+        [Theory]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]:
```suggestion
        [WinFormsTheory]
```

File: src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs [hunk 4]
```diff
@@ -2,22 +2,160 @@
...
+                TextImageRelation = value
+            };
+            Assert.Equal(value, button.TextImageRelation);
+
+            // Set same.
+            button.TextImageRelation = value;
+            Assert.Equal(value, button.TextImageRelation);
+        }
+
+        [Theory]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: ditto

File: src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs [hunk 5]
```diff
@@ -2,22 +2,160 @@
...
 using System.ComponentModel;
+using System.Drawing;
 using WinForms.Common.Tests;
+using Xunit;
 
 namespace System.Windows.Forms.Tests
 {
-    using Size = Drawing.Size;
+    using Point = System.Drawing.Point;
+    using Size = System.Drawing.Size;
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: Why do we need these? We have imported the namespace already

**Comment by RussKie** [comment]
From my observations if tests failed without an html output, tests failed b/c of an assert somewhere. Typically you should be able to replicate it locally

**Comment by hughbe** [comment]
> From my observations if tests failed without an html output, tests failed b/c of an assert somewhere. Typically you should be able to replicate it locally

Nope - a failure like #2234 
```cs
xUnit.net Console Runner v2.4.1 (64-bit .NET Core 5.0.0-alpha.1.19530.10)
  Discovering: System.Windows.Forms.Tests
  Discovered:  System.Windows.Forms.Tests
  Starting:    System.Windows.Forms.Tests
Unhandled exception. System.Exception:    at Xunit.Assert.Same(Object expected, Object actual) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\IdentityAsserts.cs:line 33
   at System.Windows.Forms.Tests.DataGridTests.<>c__DisplayClass5_0.<DataGrid_OnHandleCreated_Invoke_CallsHandleCreated>b__0(Object sender, EventArgs e) in D:\a\1\s\src\System.Windows.Forms\tests\UnitTests\System\Windows\Forms\DataGridTests.cs:line 187
   at System.Windows.Forms.Control.OnHandleCreated(EventArgs e) in D:\a\1\s\src\System.Windows.Forms\src\System\Windows\Forms\Control.cs:line 7998
   at System.Windows.Forms.DataGrid.OnHandleCreated(EventArgs e) in D:\a\1\s\src\System.Windows.Forms\src\System\Windows\Forms\DataGrid.cs:line 4133
   at System.Windows.Forms.Control.WmCreate(Message& m) in D:\a\1\s\src\System.Windows.Forms\src\System\Windows\Forms\Control.cs:line 12351
   at System.Windows.Forms.Control.WndProc(Message& m) in D:\a\1\s\src\System.Windows.Forms\src\System\Windows\Forms\Control.cs:line 13383
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m) in D:\a\1\s\src\System.Windows.Forms\src\System\Windows\Forms\Control.ControlNativeWindow.cs:line 65
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m) in D:\a\1\s\src\System.Windows.Forms\src\System\Windows\Forms\Control.ControlNativeWindow.cs:line 117
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, WindowMessage msg, IntPtr wparam, IntPtr lparam) in D:\a\1\s\src\System.Windows.Forms\src\System\Windows\Forms\NativeWindow.cs:line 375
   at System.Windows.Forms.Tests.DataGridTests.<>c.<.ctor>b__0_0(Object sender, ThreadExceptionEventArgs e) in D:\a\1\s\src\System.Windows.Forms\tests\UnitTests\System\Windows\Forms\DataGridTests.cs:line 19
   at System.Windows.Forms.Application.ThreadContext.OnThreadException(Exception t) in D:\a\1\s\src\System.Windows.Forms\src\System\Windows\Forms\Application.ThreadContext.cs:line 891
   at System.Windows.Forms.Application.OnThreadException(Exception t) in D:\a\1\s\src\System.Windows.Forms\src\System\Windows\Forms\Application.cs:line 1109
   at System.Windows.Forms.Control.WndProcException(Exception e) in D:\a\1\s\src\System.Windows.Forms\src\System\Windows\Forms\Control.cs:line 13613
   at System.Windows.Forms.Control.ControlNativeWindow.OnThreadException(Exception e) in D:\a\1\s\src\System.Windows.Forms\src\System\Windows\Forms\Control.ControlNativeWindow.cs:line 59
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, WindowMessage msg, IntPtr wparam, IntPtr lparam) in D:\a\1\s\src\System.Windows.Forms\src\System\Windows\Forms\NativeWindow.cs:line 388
=== COMMAND LINE ===
"D:\a\1\s\.dotnet\dotnet.exe" exec --depsfile "D:\a\1\s\artifacts\bin\System.Windows.Forms.Tests\Debug\netcoreapp5.0\System.Windows.Forms.Tests.deps.json" --runtimeconfig "D:\a\1\s\artifacts\bin\System.Windows.Forms.Tests\Debug\netcoreapp5.0\System.Windows.Forms.Tests.runtimeconfig.json"  "D:\a\1\s\.packages\xunit.runner.console/2.4.1/tools/netcoreapp2.0/xunit.console.dll" "D:\a\1\s\artifacts\bin\System.Windows.Forms.Tests\Debug\netcoreapp5.0\System.Windows.Forms.Tests.dll" -noautoreporters -xml "D:\a\1\s\artifacts\TestResults\Debug\System.Windows.Forms.Tests_netcoreapp5.0_x64.xml" -html "D:\a\1\s\artifacts\TestResults\Debug\System.Windows.Forms.Tests_netcoreapp5.0_x64.html"  > "D:\a\1\s\artifacts\log\Debug\System.Windows.Forms.Tests_netcoreapp5.0_x64.log" 2>&1
```

File: src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs (hunk 5 above)
Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: Because building these tests against the .NET Framework causes conflicts between System.Windows.Size and System.Drawing.Size

Comment (weltkante) [reply src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: `System.Windows.Size` is from WindowsBase.dll - are the tests declaring `<UseWPF>` or are is this now always referenced even if you don't use WPF?

Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: This is from a winforms project

Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: E.g.
```cs
mber. The new keyword is not required. [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(25,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(25,38): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(26,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(26,38): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(27,32): error CS0117: 'Point' does not contain a definition for 'Empty' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(41,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(41,46): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(82,32): error CS0117: 'Point' does not contain a definition for 'Empty' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(84,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(84,38): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(85,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(85,38): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(88,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(88,38): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(95,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(95,46): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(562,24): error CS0029: Cannot implicitly convert type 'System.Windows.Size' to 'System.Drawing.Size' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(565,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(565,44): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(577,24): error CS0029: Cannot implicitly convert type 'System.Windows.Size' to 'System.Drawing.Size' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(580,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(580,44): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(597,92): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'System.Drawing.Size' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(1034,51): error CS0029: Cannot implicitly convert type 'System.Drawing.Size' to 'System.Windows.Size' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(1036,51): error CS0029: Cannot implicitly convert type 'System.Drawing.Size' to 'System.Windows.Size' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(1040,44): error CS0029: Cannot implicitly convert type 'System.Drawing.Size' to 'System.Windows.Size' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]

Build FAILED.

PanelTests.cs(597,29): warning CS0109: The member 'PanelTests.BorderedPanel.SizeFromClientSize(Size)' does not hide an accessible member. The new keyword is not required. [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(25,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(25,38): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(26,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(26,38): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(27,32): error CS0117: 'Point' does not contain a definition for 'Empty' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(41,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(41,46): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(82,32): error CS0117: 'Point' does not contain a definition for 'Empty' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(84,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(84,38): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(85,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(85,38): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(88,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(88,38): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(95,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(95,46): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(562,24): error CS0029: Cannot implicitly convert type 'System.Windows.Size' to 'System.Drawing.Size' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(565,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(565,44): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(577,24): error CS0029: Cannot implicitly convert type 'System.Windows.Size' to 'System.Drawing.Size' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(580,26): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(580,44): error CS1503: Argument 2: cannot convert from 'System.Drawing.Size' to 'string' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(597,92): error CS1503: Argument 1: cannot convert from 'System.Windows.Size' to 'System.Drawing.Size' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(1034,51): error CS0029: Cannot implicitly convert type 'System.Drawing.Size' to 'System.Windows.Size' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(1036,51): error CS0029: Cannot implicitly convert type 'System.Drawing.Size' to 'System.Windows.Size' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
PanelTests.cs(1040,44): error CS0029: Cannot implicitly convert type 'System.Drawing.Size' to 'System.Windows.Size' [C:\Users\hughbe\source\repos\ConsoleApp\ConsoleApp.csproj]
```

Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: csproj:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net472</TargetFramework>
    <LangVersion>latest</LangVersion>
    <Product>Microsoft® .NET</Product>
    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
  </PropertyGroup>

  <ItemGroup>
    <Reference Include="System.Windows.Forms" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Moq" Version="4.13.1" />
    <PackageReference Include="xunit" Version="2.4.1" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1" />
    <PackageReference Include="Xunit.StaFact" Version="0.3.18" />
  </ItemGroup>

  <ItemGroup>
    <Content Include="TestResources\Files\*">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </Content>
    <Content Include="bitmaps\*">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </Content>
    <EmbeddedResource Include="bitmaps\10x16_one_entry_32bit.ico">
      <LogicalName>System.Windows.Forms.Design.Tests.CustomPropertyTab</LogicalName>
    </EmbeddedResource>
  </ItemGroup>

</Project>
```

Comment (weltkante) [reply src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: There must be something implicitly referencing WPF then in your build infrastructure, when I'm creating a new WinForms project `System.Windows.Size` does not resolve.

<details><summary>Project File</summary>

```xml
<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">

  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <UseWindowsForms>true</UseWindowsForms>
  </PropertyGroup>

</Project>
```

</details>

Comment (weltkante) [reply src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: > Ah see I’m using a `net472` Project

No thats not the problem, still does not resolve `System.Windows.Size` for a new project.

<details><summary>project file</summary>

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net472</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <Reference Include="System.Windows.Forms" />
  </ItemGroup>

</Project>
```

</details>

Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: Running ` dotnet build -v diag` I think this comes from `xunit.stafact`

```
                             WindowsBase
                                     NuGetIsFrameworkReference=true
                                     NuGetPackageId=Xunit.StaFact
                                     NuGetPackageVersion=0.3.18
                                     NuGetSourceType=Package
                                     Pack=false
                                     Private=false (TaskId:40)
```

File: src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs [hunk 6]
```diff
@@ -2,22 +2,160 @@
...
+            Assert.False(control.IsHandleCreated);
+        }
+
+        [WinFormsFact]
+        public void Button_CreateParams_GetDefault_ReturnsExpected()
+        {
+            using var control = new SubButton();
+            CreateParams createParams = control.CreateParams;
+            Assert.Null(createParams.Caption);
+            Assert.Equal("Button", createParams.ClassName);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: I think in this and other similar tests we should do the following:
```suggestion
            Assert.Equal(Interop.ComCtl32.WindowClasses.WC_BUTTON, createParams.ClassName);
```

File: src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs [hunk 7]
```diff
@@ -2,22 +2,160 @@
...
+            Assert.Equal(0x8, createParams.ClassStyle);
+            Assert.Equal(0, createParams.ExStyle);
+            Assert.Equal(23, createParams.Height);
+            Assert.Equal(IntPtr.Zero, createParams.Parent);
+            Assert.Null(createParams.Param);
+            Assert.Equal(0x5601000B, createParams.Style);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: If this test fails it would be hard to reason about it.
Can we substitute it with `NativeMethods.BS_ABC | NativeMethods.BS_CDE`?

File: src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs [hunk 8]
```diff
@@ -2,22 +2,160 @@
...
+
+        [WinFormsFact]
+        public void Button_CreateParams_GetDefault_ReturnsExpected()
+        {
+            using var control = new SubButton();
+            CreateParams createParams = control.CreateParams;
+            Assert.Null(createParams.Caption);
+            Assert.Equal("Button", createParams.ClassName);
+            Assert.Equal(0x8, createParams.ClassStyle);
+            Assert.Equal(0, createParams.ExStyle);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: ditto

File: src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs (hunk 6 above)
Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: I decided not to do this because I want to explicitly test the string and make sure out constants are correct. Same goes for the styles

Comment (RussKie) [reply src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: I don't think this is a correct approach. There are two different aspects here: 
1. ensure the literal value is correct, and 
2. the components reference the literal value.

If we want to test constants we should be testing `Interop.ComCtl32.WindowClasses` to ensure definitions are correct.
Testing an object other than object under test turns tests in to component/integration/etc tests, and thus immediately increase a maintenance overhead.

Suppose we follow the approach, and have a literal property which is referenced by a number of components. We write tests to ensure that all these components use the same literal (but not testing it via a reference). Then for whatever reason we need to change the value of the string, and we will have to update all these tests to reflect the change. This is not correct.

Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: But then how do we verify that the value of the string is correct?

Comment (RussKie) [reply src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: `Button` class references `Interop.ComCtl32.WindowClasses.WC_BUTTON`, and it takes the value of that const for granted:
image
So in our tests we should not assert the value of `Interop.ComCtl32.WindowClasses.WC_BUTTON`, we should just assert that the classname is the same as the const.

Asserting the value of the const is completely orthogonal to this test.

Comment (RussKie) [reply src/System.Windows.Forms/tests/UnitTests/ButtonTests.cs]: > But then how do we verify that the value of the string is correct?

If we really want to do this, we need to write tests for `Interop.ComCtl32.WindowClasses`, e.g. ``Interop.ComCtl32.WindowClassesTests` where we assert correctness of the literals.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGrid.cs [hunk 9]
```diff
@@ -4132,12 +4132,15 @@ protected override void OnHandleCreated(EventArgs e)
         {
             base.OnHandleCreated(e);
 
-            // toolTipping
-            toolTipProvider = new DataGridToolTip(this);
-            toolTipProvider.CreateToolTipHandle();
-            toolTipId = 0;
+            if (IsHandleCreated)
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGrid.cs]: Why did you add this condition?

File: src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs [hunk 10]
```diff
@@ -1033,7 +1033,6 @@ internal virtual ButtonBaseAdapter CreatePopupAdapter()
 
         internal virtual ButtonBaseAdapter CreateStandardAdapter()
         {
-            Debug.Fail("Derived classes need to provide a meaningful implementation.");
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs]: Can you clarify why you removed this assert? You don't need to remove this if you're only doing it for tests. There are other ways to deal with it:

1. Use `NoAssertContext` in the test
2. Override the method in your derived test class so it doesn't hit the assert

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGrid.cs (hunk 9 above)
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGrid.cs]: Not applicable anymore as API has been removed (but see #2235)

File: src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs (hunk 10 above)
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs]: > You don't need to remove this if you're only doing it for tests

This method is hit by public APIs and is an invalid debug assertion. Therefore we can, and should remove it. The only areas where we should use `NoAssertContext` or override the method is when testing internal APIs, which we are not doing here.

Comment (weltkante) [reply src/System.Windows.Forms/src/System/Windows/Forms/ButtonBase.cs]: > This method is hit by public APIs and is an invalid debug assertion. 

I've [noticed this](https://github.com/dotnet/winforms/pull/2399#discussion_r348536097) in another PR as well, what public API will hit this? All WinForms subclasses override it. I acknowledge that its possible to create external subclasses without overriding it, but IMHO there's value in asserting that all WinForms subclasses do override this method.

**Comment by RussKie** [comment]
Thank you
